### DISCUSSION TITLE
Need help for upgrade Indy(10) on Delphi 10.2.3

### DISCUSSION THREAD

**Discussion by varathasiva** [IndySockets/Indy#582] [Q&A]
Hi Remy,

I need one more help for upgrading to the latest Indy(10) on Delphi 10.2.3.
Last time I tried upgrading Delphi 10.2.3 Indy version to Indy(10) latest version using manual upgrade it took lot of steps to do this.
Is there any other way to do this something like upgrading using some commands?
Please help me with this.

Thanks & Regards
Siva

**Comment by rlebeau** [comment]
Upgrade details are documented at https://github.com/IndySockets/Indy/wiki/Updating-Indy There is a command-line script provided for removing the *default* Indy version from Delphi 10.2.x (`Clean_Tokyo.cmd`), but if you had already upgraded Indy before then you'll likely have to manually uninstall it.  Either way, you will still need to compile and install the 5 Indy packages manually afterwards.  There is no automated installer that does everything for you.

**Comment by rlebeau** [comment]
Do you still need help?

**Comment by waynehartley** [comment]
Hi Remy, I am trying to upgrade my Indy component set in RAD Studio XE7. I know it's old, but it is what I have. The XE7 standard version installed does not appear to support TLS1.2 event when specifically assigning it. The suggestion was to upgrade the Indy. 
I downloaded the latest from https://github.com/IndySockets/Indy. I built IndySystem210, IndyCore210 and IndyProtocols210. I built and installed dclCore210 successfully.

I built dclIndyProtocols210 successfully. When I try to install dclIndyProtocols210 I get the error "procedure entrypoint @Idglobal@IPV4ToUInt32$qqrx205System@unicodeString could not be located in the dynamic link library".

I have checked in IdGlobal.pas and the IPV4toUint32 function is there but it does not appear to be exported. Do you have a suggestion on how to resolve this?

Regards
Wayne

**Reply by rlebeau** [reply]
There is likely an older copy of `dclIndyProtocols210.bpl` still on your machine, and the IDE is trying to load that copy instead of the new one you just compiled. Make sure you have removed ALL old Indy files before installing a new version.
